core/common/test/samples/InstantSamples.kt

```diff
@@ -0,0 +1,360 @@
+/*
+ * Copyright 2019-2024 JetBrains s.r.o. and contributors.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.datetime.test.samples
+
+import kotlinx.datetime.*
+import kotlinx.datetime.format.*
+import kotlin.random.*
+import kotlin.test.*
+import kotlin.time.Duration.Companion.hours
+
+class InstantSamples {
+
+    @Test
+    fun epochSeconds() {
+        // Getting the number of whole seconds that passed since the Unix epoch
+        val instant1 = Instant.fromEpochSeconds(999_999, nanosecondAdjustment = 123_456_789)
+        check(instant1.epochSeconds == 999_999L)
+        val instant2 = Instant.fromEpochSeconds(1_000_000, nanosecondAdjustment = 100_123_456_789)
+        check(instant2.epochSeconds == 1_000_000 + 100L)
+        val instant3 = Instant.fromEpochSeconds(1_000_000, nanosecondAdjustment = -100_876_543_211)
+        check(instant3.epochSeconds == 1_000_000 - 101L)
+    }
+
+    @Test
+    fun nanosecondsOfSecond() {
+        // Getting the number of nanoseconds that passed since the start of the second
+        val instant1 = Instant.fromEpochSeconds(999_999, nanosecondAdjustment = 123_456_789)
+        check(instant1.nanosecondsOfSecond == 123_456_789)
+        val instant2 = Instant.fromEpochSeconds(1_000_000, nanosecondAdjustment = 100_123_456_789)
+        check(instant2.nanosecondsOfSecond == 123_456_789)
+        val instant3 = Instant.fromEpochSeconds(1_000_000, nanosecondAdjustment = -100_876_543_211)
+        check(instant3.nanosecondsOfSecond == 123_456_789)
+    }
+
+    @Test
+    fun toEpochMilliseconds() {
+        // Converting an Instant to the number of milliseconds since the Unix epoch
+        check(Instant.fromEpochMilliseconds(0).toEpochMilliseconds() == 0L)
+        check(Instant.fromEpochMilliseconds(1_000_000_000_123).toEpochMilliseconds() == 1_000_000_000_123L)
+        check(Instant.fromEpochSeconds(1_000_000_000, nanosecondAdjustment = 123_999_999)
+            .toEpochMilliseconds() == 1_000_000_000_123L)
+    }
+
+    @Test
+    fun plusDuration() {
+        // Finding a moment that's later than the starting point by the given amount of real time
+        val instant = Instant.fromEpochMilliseconds(epochMilliseconds = 7 * 60 * 60 * 1000)
```

Comment:
It would be more understandable to use instants obtained from `parse` and show the result as `toString` for arithmetic operation samples.

Reply:
Using `fromEpochMilliseconds` also serves to highlight the difference between `Instant` and `LocalDateTime` values. Newcomers are often confused by the distinction between them.

Reply:
Replaced with `fromEpochSeconds`/`epochSeconds + nanosecondsOfSecond`.